docs/docs/behaviors/macros.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Macro Behavior
+sidebar_label: Macros
+---
+
+## Summary
+
+The macro behavior allows configuring a sequence of other behaviors to invoke
+when the macro is pressed, including support for completing part of the sequence
+once the macro is released.
+
+## Macro Definition
+
+Each macro you want to use in your keymap gets defined first, then bound in your keymap.
+
+A macro definition looks like:
+
+```
+/ {
+    behaviors {
+        alt_tab: alt_tab_macro {
+			label = "ALT_TAB_MAC";
+			compatible = "zmk,behavior-macro";
+			#binding-cells = <0>;
+			bindings
+				= <&macro_press &kp LALT>
+				, <&macro_tap>
+				, <&kp TAB>
+                , <&macro_pause_for_release>
+				, <&macro_release &kp LSHFT>
+				;
+		};
+    };
+};
+```
+
+The macro can then be bound in your keymap by referencing it by the label `&alt_tab`.
```

Comment:
That macro is not quite the same as `&kp LA(TAB)` but also not a single-key "swapper" behavior. It is effectively `&kp LA(TAB)` when tapped, but it allows you to keep cycling forward by tapping Tab while holding the macro. On press it would cycle forward one window, then release Tab but not Alt, so you can keep holding the macro and tap Tab to cycle forward another window. 

I agree that the introductory example to press/release modes should be something simpler like holding shift while typing letters. We do need a good example for `&macro_pause_for_release` though. Can anyone come up with a better use case for that?

Reply:
Made this a simpler intro example.